double_linked_list/doubleLinkedList.c

```diff
@@ -0,0 +1,171 @@
+#include <stdio.h>
+#include <stdlib.h>
+
+struct node
+{
+	int data;
+	struct node *next;
+	struct node *prev;
+};
+
+struct node *head = NULL, *last = NULL;
+
+struct node *createNode(int key)
+{
+	head = (struct node *)malloc(sizeof(struct node));
+	head->data = key;
+	head->next = NULL;
+	head->prev = NULL;
+	last = head;
+	return head;
+}
+
+struct node *additionOfNodeAtBeginning( int key)
+{
+	struct node *temp;
+	temp = (struct node *)malloc(sizeof(struct node));
+	temp->data = key;
+	temp->next = NULL;
+	temp->next = head;
+	temp->prev = NULL;
+	head->prev = temp;
+	head = temp;
+	return head;
+}
+
+struct node *additionOfNodeAtEnding( int key)
+{
+	struct node *temp;
+	temp = (struct node *)malloc(sizeof(struct node));
+	temp->data = key;
+	temp->next = NULL;
+	temp->prev = last;
+	last->next = temp;
+	temp->next = NULL;
+	last = temp;
+	return last;
+}
+
+struct node *additionOfNodeAtRandomeIndex( int key, int index)
```

Comment:
addNodeAtGivenIndex fit better as function name .

Reply:
@ishanksoni sir, sir this is pr for double ended queue and you reviewed double linked list. I will do the required changes for double linked list  and please review this pr again.

Reply:
@ishanksoni sir, this linked list file was inserted by mistake and I am sorry for this. I have already a pr for double linked list #746 which also requires your review. I will remove this file from this pr because otherwise, this pr will solve 2 issues which is against the rules of this contributathon.